core/src/executor/evaluate/mod.rs

```diff
@@ -68,12 +68,20 @@ pub async fn evaluate<'a, T>(
             }
             .map(Evaluated::from)
         }
-        Expr::Subquery(query) => select(storage, query, context.as_ref().map(Rc::clone))
-            .await?
-            .map_ok(|row| row.take_first_value().map(Evaluated::from))
-            .next()
-            .await
-            .unwrap_or_else(|| Err(EvaluateError::NestedSelectRowNotFound.into()))?,
+        Expr::Subquery(query) => {
+            let mut stream = select(storage, query, context.as_ref().map(Rc::clone))
+                .await?
+                .map_ok(|row| row.take_first_value().map(Evaluated::from));
+            let first = stream
+                .next()
+                .await
+                .unwrap_or_else(|| Err(EvaluateError::NestedSelectRowNotFound.into()))?;
+            let second = stream.next().await;
+            match second {
+                Some(_) => Err(EvaluateError::MoreThanOneRowReturned.into()),
+                None => first,
+            }
```

Comment:
Thanks!

In here, `take(2)` would be useful in this case to simplify the codes.
```rust
select(storage, query, context.as_ref().map(Rc::clone))
    .await?
    .map_ok(|row| row.take_first_value().map(Evaluated::from))
    .take(2)
    .await
    .try_collect::<Vec<..
```